ios/NativeBridge.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  NativeBridge.swift
+//  A0Auth0
+//
+//  Created by Poovamraj T T on 15/06/23.
+//  Copyright © 2023 Facebook. All rights reserved.
+//
+
+import Auth0
+import Foundation
+import LocalAuthentication
+
+@objc
+public class NativeBridge: NSObject {
+    
+    
+    static let accessTokenKey = "accessToken";
+    static let idTokenKey = "idToken";
+    static let expiresAtKey = "expiresAt";
+    static let scopeKey = "scope";
+    static let refreshTokenKey = "refreshToken";
+    static let typeKey = "type";
+    static let tokenTypeKey = "tokenType";
+    static let expiresInKey = "expiresIn";
+    static let dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+    
+    static let errorCode = "a0.invalid_state.credential_manager_exception"
+    
+    var credentialsManager: CredentialsManager
+    var clientId: String
+    var domain: String
+    
+    @objc public init(clientId: String, domain: String) {
+        let auth0 = Auth0
+            .authentication(clientId: clientId, domain: domain)
+        self.clientId = clientId
+        self.domain = domain
+        self.credentialsManager = CredentialsManager(authentication: auth0)
+        super.init()
+   }
+    
+    @objc public func webAuth(scheme: String?, state: String?, nonce: String?, audience: String?, scope: String?, connection: String?, maxAge: Int, organization: String?, invitationUrl: String?, leeway: Int, ephemeralSession: Bool, additionalParameters: [String: String], resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
+        let builder = Auth0.webAuth(clientId: self.clientId, domain: self.domain)
+        if let value = nonce {
+            let _ = builder.nonce(value)
+        }
+        if let value = audience {
+            let _ = builder.audience(value)
+        }
+        if let value = scope {
+            let _ = builder.scope(value)
+        }
+        if let value = connection {
+            let _ = builder.connection(value)
+        }
+        if(maxAge != 0) {
+            let _ = builder.maxAge(maxAge)
+        }
+        if let value = organization {
+            let _ = builder.organization(value)
+        }
+        if let value = invitationUrl, let invitationURL = URL(string: value) {
+            let _ = builder.invitationURL(invitationURL)
+        }
+        if(leeway != 0) {
+            let _ = builder.leeway(leeway)
+        }
+        if(ephemeralSession) {
+            let _ = builder.useEphemeralSession()
+        }
+        builder.parameters(additionalParameters)
+        builder.start { result in
+            switch result {
+            case .success(let credentials):
+                resolve(credentials.asDictionary())
+            case .failure(let error):
+                reject(NativeBridge.errorCode, error.errorDescription, error)
```

Comment:
Won't this use `a0.invalid_state.credential_manager_exception` as the error code for Web Auth login?

Reply:
Note that Web Auth operations produce a `WebAuthError`, which have different cases, according to what the error is: https://auth0.github.io/Auth0.swift/documentation/auth0/webautherror

In the Flutter SDK we map the cases to unique error codes: https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/ios/Classes/WebAuth/WebAuthExtensions.swift#L4